src/main/java/io/vertx/core/json/JsonObject.java

```diff
@@ -88,6 +88,26 @@ public JsonObject(Buffer buf) {
     }
   }
 
+  /**
+   * Create a JsonObject from some mapping pairs from arguments.
+   * Ignore the last argument if the number of arguments is not even.
+   *
+   * @param args The key/value pairs.
+   */
+  public static JsonObject of(Object ...args) {
+    JsonObject obj = new JsonObject();
```

Comment:
Maybe some validation should be performed here? If the number of elements is not even, throw `IllegalArgumentException`?

Reply:
And instead of creating immediately the `JsonObject`, how about creating a map of the right size:

```java
Map map = new HashMap(length/2);
```

Populate the map and then eventually:

```java
return new JsonObject(map);
```